### PR TITLE
trayscale: add module

### PR DESCRIPTION
**Pull request by callumio**
### Description

Trayscale is an unofficial GUI wrapper around the Tailscale CLI client.

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all` or `nix develop --ignore-environment .#all` using Flakes.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://nix-community.github.io/home-manager/#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [x] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

### CODE REVIEW

File: modules/services/trayscale.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let cfg = config.services.trayscale;
+in {
+  meta.maintainers = [ hm.maintainers.callumio ];
+
+  options.services.trayscale = {
+    enable = mkEnableOption
+      "An unofficial GUI wrapper around the Tailscale CLI client.";
```
Comment (sumnerevans) [inline modules/services/trayscale.nix:9]:
```suggestion
      "trayscale, an unofficial GUI wrapper around the Tailscale CLI client";
```

File: modules/services/trayscale.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    package = mkPackageOption pkgs "trayscale" { };
+    hideWindow = mkOption {
+      description = "Whether to hide the trayscale window on startup.";
+      type = types.bool;
+      default = true;
+    };
```
Comment (sumnerevans) [inline modules/services/trayscale.nix:15]:
```suggestion
    hideWindow = mkEnableOption "hide the trayscale window on startup" // {
      default = true;
    };
```

File: tests/modules/services/trayscale/hide-window.nix [hunk 3]
```diff
@@ -0,0 +1,17 @@
+{ config, pkgs, ... }: {
+  config = {
+    services.trayscale = {
+      enable = true;
+      hideWindow = true;
+    };
+
+    test.stubs = { trayscale = { }; };
```
Comment (sumnerevans) [inline tests/modules/services/trayscale/hide-window.nix:8]:
```suggestion
    test.stubs.trayscale = { };
```

File: tests/modules/services/trayscale/show-window.nix [hunk 4]
```diff
@@ -0,0 +1,17 @@
+{ config, pkgs, ... }: {
+  config = {
+    services.trayscale = {
+      enable = true;
+      hideWindow = false;
+    };
+
+    test.stubs = { trayscale = { }; };
```
Comment (sumnerevans) [inline tests/modules/services/trayscale/show-window.nix:8]:
```suggestion
    test.stubs.trayscale = { };
```

**Review by sumnerevans — CHANGES_REQUESTED** [review]

**Review by sumnerevans — APPROVED** [review]
Please rebase and squash. Hopefully that will make at least the Ubuntu tests pass.

**Comment by rycee** [comment]
Thanks! Rebased to master in 4c8647b1ed35d0e1822c7997172786dfa18cd7da 🙂

**Pull request closed without merging**